### PR TITLE
add authorization rules

### PR DESCRIPTION
**Pull request by Ibizugbe**
Hello👋,
In this PR, I have completed the current milestone of the project and adhered to the project's requirements.
- no linter errors.
- the project is professionally documented.
- the right git-flow method has been used.
- CanCanCan has been installed in the project.
- a role column has been added to the user table using a migration.
- an authorization has been created using CanCanCan that gives an admin or a user(specific) permission to delete a post.
- an authorization has been created using CanCanCan that gives an admin or a user(specific) permission to delete a comment.
Thanks for your time🤗🤗👋👍

**NOTE**
In case, you see this error 👉Net::SMTPServerBusy in Devise::RegistrationsController#create. Kindly know that it is a window error. You will see the confirmed password on your console, kindly find the attached link  video that I have made 👉 [vdeo](https://www.loom.com/share/3047dd7d421142ffa92badc07838471d)

### CODE REVIEW

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] Kindly notice that you have submitted the wrong pull request this submission should be for the API so kindly submit the right pull request.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
- [ ] I am getting an error when opening the browser I can't signin, please fix that
Screenshot from 2022-11-03 18-30-26




### Cheers!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @Ibizugbe, 

Great work on making the changes requested by a previous reviewer 👏 🔥
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights 🎉🔥
- Set the `ability` class that defines the set of rules for user - post interactions. ✔
- Linters ✔

 ## Required Changes ♻️
- [ ] The project is not running properly. Check the image below

Screen Shot 2022-11-03 at 6 45 38 PM

_Check the comments under the review._

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lizdev-05** [comment]
You are supposed to tell us why the project is not opening. I guess you are trying to log in when you do not have an account. You are supposed to register 😒

**Comment by EricMbouwe** [comment]
It is not at the login stage but when the app is launched for the first time. As you can see in the image there is a file which is not resolved. You can check the full trace to have a better understanding of the issue.

**Comment by Lizdev-05** [comment]
@EricMbouwe ,  Kindly check this video that I recorded for that purpose... 👉[video](https://www.loom.com/share/3047dd7d421142ffa92badc07838471d)

**Comment by EricMbouwe** [comment]
I guess the issue could be due to the system, I am using a mac. Since I can not re-review your project without a new submission, I would suggest requesting another review linking that video for the purpose.

**Comment by Lizdev-05** [comment]
Okay @EricMbouwe , Thank you.

File: app/models/user.rb [hunk 1]
```diff
@@ -14,4 +14,8 @@ class User < ApplicationRecord
   def return_recent_posts
     posts.order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (MahmoudBakr23) [inline app/models/user.rb:20]: Highlights
- Well done creating the `admin?` helper method in the User model. 💯

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done installing cancan gem and setting up the Ability model
- Great job modifying your User model for that matter
- Well done using `load_and_authorize_resource` in your posts controller
- Great communication with the previous reviewers

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Lizdev-05) [reply app/models/user.rb:20]: Thank you for reviewing and approving our project.

Comment (MahmoudBakr23) [reply app/models/user.rb:20]: You're welcome, good luck with the next 😄

**Pull request merged**